xcdat/spatial_avg.py

```diff
@@ -324,93 +324,155 @@ def _get_weights(
-
-            if domain_bounds is not None:
-                dom_bounds = domain_bounds.copy()
-
-                # validate domain bounds
-                self._validate_domain_bounds(dom_bounds)
-
-                if region_bounds is not None:
-                    reg_bounds = np.array(region_bounds).astype(float)
-                    if dim == "lon":
-                        # Ensure that domain and regional bounds are both
-                        # in the range of 0 -> 360
-                        dom_bounds = self._swap_lon_axes(dom_bounds, to=360)
-                        reg_bounds = self._swap_lon_axes(reg_bounds, to=360)
-
-                        # Some grid cells cross the prime meridian (
-                        # e.g., [-1, 1]). Check for this special case. If it
-                        # applies, return the index of that grid cell
-                        # (otherwise None) and recreate the dims with two
-                        # grid cells ([0, 1] and [359, 360]).
-                        pmb_index, dom_bounds = self._align_longitude_to_360_axis(
-                            dom_bounds
-                        )
-
-                        # If the region bounds are equal, it means all
-                        # longitude values are specified (e.g., (360, 360) ==
-                        # (0, 360)). As a result, set the region lon bounds to
-                        # [0, 360].
-                        if reg_bounds[1] - reg_bounds[0] == 0:
-                            reg_bounds[0] = 0.0
-                            reg_bounds[1] = 360.0
-
-                    # Scale the domain boundaries to the regional domain
-                    dom_bounds = self._scale_domain_to_region(dom_bounds, reg_bounds)
-
-                if dim == "lat":
-                    # The area between two lines of latitude scales as the
-                    # difference of the sine of latitude bounds. This scaling
-                    # is applied here.
-                    dom_bounds = np.sin(np.radians(dom_bounds))
-                    pmb_index = None
-
-                # Perform differencing to get weights and add metadata.
-                dim_wts = np.abs(dom_bounds[:, 1] - dom_bounds[:, 0])
-
-                # If pmb_index is not None it means that an extra
-                # longitude bound was added and the weight vector
-                # is too long. This extra weight should be added to the
-                # grid cell that crossed the prime meridian (and then
-                # removed from the dim_wts vector).
-                if pmb_index is not None:
-                    dim_wts[pmb_index] = dim_wts[pmb_index] + dim_wts[-1]
-                    dim_wts = dim_wts[0:-1]
-
-                # ensure attributes remain on dim_wts
-                dim_wts.attrs = dom_bounds.attrs
-                dim_wts.name = dim + "_wts"
-                axis_weights[dim] = dim_wts
+        for axes, bounds_by_type in bounds.items():
```

Comment:
@pochedls  I just noticed that `get_weights()` calculates weights for all axes even if they aren't specified in the `axis` argument.

I'm adding code that limits calculation to only axes that are relevant to the user.